src/backend/pipeline/modules/module.py

```diff
@@ -76,3 +79,30 @@ def prepare(self, data: Data):
         """Prepares the module to be run."""
 
         data.modules[self.type] = {}
+
+    def upload(self, bucket_name: str, data: Data):
+        """Upload data to Google Storage."""
+        
+        base_url: str = "https://storage.cloud.google.com/" + bucket_name + "/"
+        try:
+            # connect to firestore
+            db = firestore.client()
+            # get 'pipelines' collection
+            collection = db.collection('pipelines')
+            storage_client = storage.Client() 
+            bucket = storage_client.bucket(bucket_name)
```

Comment:
I feel like creating a client for every upload is overkill. It'd be better to move the pipeline-specific Cloud Store data (client object, bucket path...) to the `Pipeline` itself. Then, you don't need to `upload` for every module/runnable (making it easier to deal with `ParallelModules`), avoiding an `async` hell.